### ISSUE TITLE
Incorrect Spring Java code generation?

### ISSUE THREAD

**Issue by RichardBruskiewich** [swagger-api/swagger-codegen#9644]
##### Description

**Incorrect code snippet:**` ResponseEntity<Map<String, Map<String>>` emitted instead of `ResponseEntity<Map<String, Map<String>>`

Here's an example snippet of the kind of code (associated YAML snippet of spec shown below)

```
@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.SpringCodegen", date = "2019-08-15T17:03:28.599Z[GMT]")
@Api(value = "predicates", description = "the predicates API")
public interface PredicatesApi {

    @ApiOperation(value = "Get supported relationships by source and target", nickname = "predicatesGet", notes = "", response = Map.class, responseContainer = "Map", tags={  })
    @ApiResponses(value = { 
        @ApiResponse(code = 200, message = "Predicates by source and target", response = Map.class, responseContainer = "Map") })
    @RequestMapping(value = "/predicates",
        produces = { "application/json" }, 
        method = RequestMethod.GET)
    ResponseEntity<Map<String, Map<String>> predicatesGet();

}

```

Along with the above error, the associated Controller code also improperly escapes the double quotes of the JSON in the example output (see the *objectMapper.readValue* input parameter below):

```
    public ResponseEntity<Map<String, Map<String>> predicatesGet() {
        String accept = request.getHeader("Accept");
        if (accept != null && accept.contains("application/json")) {
            try {
                return new ResponseEntity<Map<String, Map<String>>(objectMapper.readValue("{
  "chemical_substance" : {
    "gene" : [ "directly_interacts_with", "decreases_activity_of" ]
  }
}", Map.class), HttpStatus.NOT_IMPLEMENTED);
            } catch (IOException e) {
                log.error("Couldn't serialize response for content type application/json", e);
                return new ResponseEntity<Map<String, Map<String>>(HttpStatus.INTERNAL_SERVER_ERROR);
            }
        }

        return new ResponseEntity<Map<String, Map<String>>(HttpStatus.NOT_IMPLEMENTED);
    }
```
##### Swagger-codegen version

3.0.10

##### Swagger declaration file content or url

```yaml
/predicates:
    get:
      summary: Get supported relationships by source and target
      responses:
        '200':
          description: Predicates by source and target
          content:
            application/json:
              schema:
                description: Source map
                type: object
                additionalProperties:
                  description: Target map
                  type: object
                  additionalProperties:
                    description: Array of predicates
                    type: array
                    items:
                      type: string
                example:
                  chemical_substance:
                    gene:
                      - directly_interacts_with
                      - decreases_activity_of
```

##### Command line used for generation

https://app.swaggerhub.com/apis *Export* function

##### Steps to reproduce

Downloaded codegen generated zip file then attempted build.

**Comment by elier** [comment]
Same problem:

java --version
java 11.0.4 2019-07-16 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.4+10-LTS)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.4+10-LTS, mixed mode)

mvn --version
Maven home: /usr/local/Cellar/maven/3.6.1/libexec
Java version: 11.0.4, vendor: Oracle Corporation, runtime: /Library/Java/JavaVirtualMachines/jdk-11.0.4.jdk/Contents/Home
Default locale: en_CA, platform encoding: UTF-8
OS name: "mac os x", version: "10.14.6", arch: "x86_64", family: "mac"

**Comment by elier** [comment]
I've test all previous versions and the problem is present in v3.0.9 as well.

curl http://central.maven.org/maven2/io/swagger/codegen/v3/swagger-codegen-cli/3.0.9/swagger-codegen-cli-3.0.9.jar -o swagger-codegen-cli.jar

java -jar ./swagger-codegen-cli.jar generate -i http://petstore.swagger.io/v2/swagger.json -l spring -o ../petstore_api

Not present in v.3.0.8:
curl http://central.maven.org/maven2/io/swagger/codegen/v3/swagger-codegen-cli/3.0.8/swagger-codegen-cli-3.0.8.jar -o swagger-codegen-cli.jar

java -jar ./swagger-codegen-cli.jar generate -i http://petstore.swagger.io/v2/swagger.json -l spring -o ../petstore_api

**Comment by RichardBruskiewich** [comment]
I just manually patched up the code to eventually make it work.

I'm just not familiar with the codegen code base, and don't have too much time right now, but I'd otherwise be tempted to dive in there and try to fix things. I wonder where I'd start looking for the relevant code snippets?

**Comment by frantuma** [comment]
Thanks for reporting this! all issues mentioned are fixed in latest snapshot and upcoming 3.0.11 release, closing ticket, please reopen if you're still experiencing issues
